backend/rates.go

```diff
@@ -44,26 +44,34 @@ var (
 // GetRatesUpdaterInstance gets a singleton instance of RatesUpdater.
 func GetRatesUpdaterInstance() *RatesUpdater {
 	ratesUpdaterInstanceOnce.Do(func() {
-		ratesUpdaterInstance = NewRatesUpdater()
+		ratesUpdaterInstance = &RatesUpdater{
+			last:       map[string]map[string]float64{},
+			log:        logging.Get().WithGroup("rates"),
+			socksProxy: socksproxy.NewSocksProxy(false, ""),
+		}
 	})
 	return ratesUpdaterInstance
 }
 
 // RatesUpdater implements coin.RatesUpdater.
 type RatesUpdater struct {
 	observable.Implementation
-	last map[string]map[string]float64
-	log  *logrus.Entry
+	last       map[string]map[string]float64
+	log        *logrus.Entry
+	socksProxy socksproxy.SocksProxy
 }
 
 // NewRatesUpdater returns a new rates updater.
-func NewRatesUpdater() *RatesUpdater {
-	updater := &RatesUpdater{
-		last: map[string]map[string]float64{},
-		log:  logging.Get().WithGroup("rates"),
-	}
-	go updater.start()
-	return updater
+func NewRatesUpdater(socksProxy socksproxy.SocksProxy) *RatesUpdater {
+	ratesUpdaterInstanceOnce.Do(func() {
```

Comment:
Please remove the global now that it is not a real global var anymore.

The other option would be to have one global per possible parameter input to this function, but that would be silly ;)